state.go

```diff
@@ -242,6 +243,21 @@ func (m *Memberlist) probeNodeByAddr(addr string) {
 	m.probeNode(n)
 }
 
+// failedRemote checks the error and decides if it indicates a failure on the
+// other end.
+func failedRemote(err error) bool {
```

Comment:
maybe add a test of this function for sanity?

Reply:
absolutely! incoming...

Reply:
done